tests/fixtures/mariadb_fixtures.py

```diff
@@ -79,7 +79,11 @@ def mariadb_example_test_dataset_config(
 def mariadb_integration_session(connection_config_mariadb):
     example_mariadb_uri = MariaDBConnector(connection_config_mariadb).build_uri()
     engine = get_db_engine(database_uri=example_mariadb_uri)
-    SessionLocal = get_db_session(engine=engine)
+    SessionLocal = get_db_session(
+        engine=engine,
+        autocommit=True,
+        autoflush=True,
```

Comment:
That's a good question — by default SQLAlchemy won't commit the `.excute(...)` transactions to the DB unless you explicitly call `session.commit()`. This means that we either need to call it manually, or set `autocommit=True` on the session. What happens if we don't call `commit()` in some form is that you end up with the session from the `mariadb_integration_db` fixture having access to all the data (from the uncommitted transactions), while the DB session that the appplication uses does not, which can be very confusing to pick apart, because the test access requests will return nothing as though the data isn't there, while the fixture's session shows the data present.

Reply:
ah got it, thank you ⭐